src/nvim/buffer.c

```diff
@@ -4528,16 +4528,20 @@ chk_modeline (
           e = s + 4;
         else
           e = s + 3;
-        vers = getdigits_int(&e);
+        if (getdigits_safe(&e, &vers) < 0) {
```

Comment:
Also regarding the specific code here, it should check `!= OK`. I fixed this during the merge.